index.js

```diff
@@ -44,10 +44,15 @@ function parse(str) {
 
   var seg = obj.path.split('/').filter(Boolean);
   var hasBlob = seg[2] === 'blob';
-  if (hasBlob && !isChecksum(seg[3])) {
-    obj.branch = seg[3];
-    if (seg.length > 4) {
-      obj.filepath = seg.slice(4).join('/');
+  if (hasBlob) {
+    if (isChecksum(seg[3])) {
+      obj.branch = 'master'
```

Comment:
what if the repo's default branch has a different name? `main` is common now, and it's also fully configurable.

Reply:
Yeah, doesn't makes sense now that I look at it again.

It may be possible to determine the branch with a reverse ref search, but not solely from the url.